riscv-runtime/src/hash.rs

```diff
@@ -27,3 +27,85 @@ pub fn poseidon_gl(data: [u64; 12]) -> [u64; 4] {
 
     poseidon_gl_unsafe(data)
 }
+
+/// Calls the keccakf machine
+/// Return value is placed in the same array as the input
+/// This is unsafe because it does not check if the u64 elements fit the Goldilocks field.
+pub fn keccakf_unsafe(mut data: [u64; 25]) -> [u64; 25] {
```

Comment:
Maybe it makes sense to take the input by reference, allocate a new array and return that?

Reply:
Yeah, these values are quite large, maybe it should take a slice and return a `Vec`?

Reply:
Or maybe take two parameters? `input` as an immutable slice, and `output` as a mutable slice (then the caller can decide how to allocate space).

Reply:
The keccak loop could allocate two arrays and alternate them as input/output for example.

Reply:
Makes sense. Thanks!

Reply:
Done.